src/dune_rules/lib.ml

```diff
@@ -834,17 +852,18 @@ end = struct
   let instantiate_impl (db, name, info, hidden) =
     let open Memo.O in
     let unique_id = Id.make ~name ~path:(Lib_info.src_dir info) in
-    let status = Lib_info.status info in
     let private_deps =
-      match status with
+      match Lib_info.status info with
       (* [Allow_all] is used for libraries that are installed because we don't
          have to check it again. It has been checked when compiling the
          libraries before their installation *)
-      | Installed_private | Private (_, None) | Installed -> Allow_all
+      | Installed_private | Private (_, None) | Installed ->
+        Private_deps.Allow_all
       | Private (_, Some _) -> From_same_project `Private_package
       | Public (_, _) -> From_same_project `Public
     in
     let resolve name = resolve_dep db name ~private_deps in
+    let for_lib = Some info in
```

Comment:
I find it a bit weird that we need this parameter at all. I would expect us to push the stackframe that corresponds to it and then resume the work as normal

Reply:
OK. I might be misunderstanding how it all works, but they seem to be added after resolution here: https://github.com/ocaml/dune/blob/bfbc453f5d47fcf2b50ec3260e0252900651cfba/src/dune_rules/lib.ml#L957-L962

I tried to preserve the same behavior.